generator/src/main/java/com/scottlogic/deg/generator/decisiontree/serialisation/DecisionTreeMapper.java

```diff
@@ -19,17 +19,7 @@ public DecisionTree fromDto(DecisionTreeDto decisionTreeDto) {
             getMappedProfileFields(decisionTreeDto),
             decisionTreeDto.description);
     }
-    
-    // Pair A2 
-    public DecisionTreeDto toDto(DecisionTree tree) {
-        DecisionTreeDto dto = new DecisionTreeDto();
-        dto.rootNode = toDto(tree.getRootNode());
-        dto.fields = tree
-                .fields.stream().map(f -> new FieldDto(f.name)).collect(Collectors.toList());
-        dto.description = tree.description;
-        return dto;
-    }
-    
+
```

Comment:
I'd keep this in place, personally. We have the deserialise methods so makes sense the keep the serialise equivalents

Reply:
makes sense.

though i'm thinking we should maybe get rid of the serialise. as it's only used in tests for optimiseing tree, and partitioning tree.

both of which can be done without deserialising the tree

Reply:
Not sure I follow.

Not the partitioner and optimiser take a tree as the input and emit a tree as the output. This is what you'd want to provide and verify against. Therefore deserialise is required for provide the input to the test and serialise is required to assert the result matches the expected result.

You could get around it by using deserialise only, but this would then mean you'd have to hand craft the 'expected output' which is time consuming and error prone, hence the existence of the serialise method.

Reply:
there is a constraintNodeBuilder that we could use for input and expected tree. the tests would be easier to follow this way too.

Reply:
My feeling is we should keep this in even though it doesn't seem to be used at present.

What does the comment "Pair A2" refer to?

Reply:
re added the method